andela/cyclops-backend
# #167728046 Complete user can book accommodation facility

**What does this PR do?**

- Implement endpoint `POST /bookings` for user to book an accommodation
- Create an endpoint `GET /bookings` for user to see all bookings

  **Description of Task to be completed?**

- Login with `Supplier` role
- Create an accommodation and a room at the endpoint `POST /accommodation`
- Use the new or existing room `UUID` to create a booking

  **How should this be manually tested?**

- Clone the project and seed the database
- Log in as a seeded user 
- Visit `/bookings POST` request and enter the parameters:
- `room_uuid` (it must be a UUID of an existing room)
- `starting_date`
- `ending_date`

To see all bookings, visit `GET /bookings` as a logged-in user (with token)

  **Any background context you want to provide?**

  A user should be able to book accommodation at his/her destination of traveling to be assured of residence on getting there.

  **What are the relevant stories?**

[#167728046](https://www.pivotaltracker.com/story/show/167728046)

  **Screenshots (if appropriate)**

<img width="906" alt="create booking" src="https://user-images.githubusercontent.com/11611867/64900680-8c593c00-d68a-11e9-869d-2449d682f409.PNG">

<img width="864" alt="get all booking" src="https://user-images.githubusercontent.com/11611867/64900686-97ac6780-d68a-11e9-945f-621b6584690e.PNG">
